### PR TITLE
Add logic for user login series.

### PR DESCRIPTION
**Pull request by WojciechOsak**

### CODE REVIEW

File: src/main/java/pl/brotherhood/services/PointsService.java [hunk 1]
```diff
@@ -26,4 +43,42 @@ public PointsEntity addPointsEntity() {
...
+    }
+
+    private void completeSeries(PointsConfigurationEntity pointsConfiguration, PointsEntity pointsEntity) {
+        pointsRepository.resetDaysSeries(pointsEntity);
+        pointsRepository.increaseSeriesCompleted(pointsEntity);
+        pointsRepository.addPoints(pointsEntity, pointsConfiguration.getSeriesReward());
+       // pointsRepository.updateLastLogDate(pointsEntity, new java.sql.Date(System.currentTimeMillis()));
+    }
+
+    public PointsEntity getPointsEntityByUserPassId(@NonNull Integer userPassId) {
```
Comment (gosak) [inline src/main/java/pl/brotherhood/services/PointsService.java]: I was using @Nullable, i think the better idea is to use one annotation for that.

File: src/main/java/pl/brotherhood/repositories/impl/PointsRepositoryImpl.java [hunk 2]
```diff
@@ -17,7 +20,41 @@ public void addPointsEntity(PointsEntity pointsEntity) {
...
+    public void resetDaysSeries(PointsEntity pointsEntity) {
+        entityManager.merge(pointsEntity.setDaysCompleted(0));
+    }
+
+    @Override
+    public PointsEntity getUserPoints(Integer userPassId) {
+        List usersList = entityManager.createQuery("SELECT u FROM UserEntity u" +
+                " WHERE u.userPass.id = :param ")
+                .setParameter("param", userPassId)
+                .getResultList();
```
[the remark this answers is not in the document]
Comment (gosak) [reply src/main/java/pl/brotherhood/repositories/impl/PointsRepositoryImpl.java]: And whole query can be like:
createQuery("SELECT u.pointsEntity FROM UserEntity u ...")..().getSingleResult()

File: src/main/java/pl/brotherhood/repositories/impl/UserPassRepositoryImpl.java [hunk 3]
```diff
@@ -16,12 +18,14 @@ public void addUserPassEntity(UserPassEntity userPassEntity) {
...
-        return entityManager.createQuery("SELECT u FROM UserPassEntity u " +
-                "WHERE u.username LIKE :username AND u.passwordHash LIKE :passwordHash")
+    public UserPassEntity getUserPass(String username, String passwordHash) {
+        List usersList = entityManager.createQuery("SELECT u FROM UserPassEntity u" +
+                " WHERE u.username LIKE :username " +
+                "AND u.passwordHash LIKE :passwordHash")
                 .setParameter("username", username)
                 .setParameter("passwordHash", passwordHash)
-                .getResultList().size() > 0;
+                .getResultList();
```
Comment (gosak) [inline src/main/java/pl/brotherhood/repositories/impl/UserPassRepositoryImpl.java]: same as above

File: src/main/java/pl/brotherhood/repositories/impl/AbstractRepository.java [hunk 4]
```diff
@@ -5,11 +5,13 @@
 package pl.brotherhood.repositories.impl;
 
 import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
 
 import javax.persistence.EntityManager;
 import javax.persistence.PersistenceContext;
 
 @Repository
+@Transactional
```
[the remark this answers is not in the document]
Comment (WojciechOsak) [reply src/main/java/pl/brotherhood/repositories/impl/AbstractRepository.java]: Yes - they are extended by childs. 
Without transactional we can't update any entity in database.

File: src/main/java/pl/brotherhood/repositories/impl/AbstractRepository.java [hunk 5]
```diff
@@ -5,11 +5,13 @@
...
 public abstract class AbstractRepository {
     @PersistenceContext
     EntityManager entityManager;
```
[the remark this answers is not in the document]
Comment (WojciechOsak) [reply src/main/java/pl/brotherhood/repositories/impl/AbstractRepository.java]: That was your job :D But ok I will change it in this review.

File: src/main/java/pl/brotherhood/repositories/impl/PointsRepositoryImpl.java [hunk 6]
```diff
@@ -17,7 +20,41 @@ public void addPointsEntity(PointsEntity pointsEntity) {
     }
 
     @Override
-    public PointsEntity getUserPoints(Integer userId) {
-        return entityManager.find(UserEntity.class, userId).getPointsEntity();
+    public void addPoints(PointsEntity pointsEntity, Integer points) {
+        PointsEntity updatedPointsEntity = entityManager.find(PointsEntity.class, pointsEntity.getId());
```
[the remark this answers is not in the document]
Comment (WojciechOsak) [reply src/main/java/pl/brotherhood/repositories/impl/PointsRepositoryImpl.java]: Good point. My fault, sorry.

File: src/main/java/pl/brotherhood/repositories/impl/AbstractRepository.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (gosak) [reply src/main/java/pl/brotherhood/repositories/impl/AbstractRepository.java]: There is no reason to give it for whole class, simple functions doesn't need it.

File: src/main/java/pl/brotherhood/repositories/impl/PointsRepositoryImpl.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (WojciechOsak) [reply src/main/java/pl/brotherhood/repositories/impl/PointsRepositoryImpl.java]: Thanks I will change it.

File: src/main/java/pl/brotherhood/repositories/impl/AbstractRepository.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (gosak) [reply src/main/java/pl/brotherhood/repositories/impl/AbstractRepository.java]: Im sure that @Repository is not extended by childs (in example remove @Repository from one of implementation -> spring cannot autowire it), so i think it's same with @Transactional

File: src/main/java/pl/brotherhood/repositories/impl/AbstractRepository.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (gosak) [reply src/main/java/pl/brotherhood/repositories/impl/AbstractRepository.java]: i know :D

File: src/main/java/pl/brotherhood/repositories/impl/AbstractRepository.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (WojciechOsak) [reply src/main/java/pl/brotherhood/repositories/impl/AbstractRepository.java]: Wihout transactional we can't update anything in database - just try it without it and you will see :)

[the remark this answers is not in the document]
Comment (gosak) [reply src/main/java/pl/brotherhood/repositories/impl/AbstractRepository.java]: I know, but you can use @Transactional on functions. If you are use it for whole class then every function "has annotation above"

File: src/main/java/pl/brotherhood/builders/impl/PointsEntityFactory.java [hunk 7]
```diff
@@ -21,7 +23,8 @@ public PointsEntity build(AbstractRequest abstractRequest) {
     public PointsEntity build() {
         return new PointsEntity()
                 .setAmount(0)
-                .setLogSeries(1)
-                .setLastLog(new Date(System.currentTimeMillis()));
+                .setSeriesCompleted(0)
+                .setDaysCompleted(0)
+                .setLastLog(new Date(System.currentTimeMillis() - ONE_DAY));
```
Comment (gosak) [inline src/main/java/pl/brotherhood/builders/impl/PointsEntityFactory.java]: Why one day before register?

**Pull request merged**